### ISSUE TITLE
difference

### ISSUE THREAD

**Issue by chenbolinstudent** [AceCoooool/DSS-pytorch#12]
what is the difference between your code and the orginal paper :Deeply Supervised Salient Object Detection with Short Connections ？
thanks

**Comment by AceCoooool** [comment]
official version: [Andrew-Qibin](https://github.com/Andrew-Qibin/DSS)
1. training stragedy. please see [solver.prototxt](https://github.com/Andrew-Qibin/DSS/blob/master/solver.prototxt) --- I use simple adam with larger learning rate and without weight decay. （I think this may cause difference）
2. The combine way: training phase, the official use "learnable combine weights"([this line](https://github.com/Andrew-Qibin/DSS/blob/5551909f84f245009fcf78bc42d942bb4de86548/deploy.prototxt#L290)) , and the v2 in here use this stragedy, and v1 just use average stragdy. （However, they use only 3 layers for reference, which I am a littile confusing.）

The mainly gap I think is caused by learning stragedy. （Like init and learning rate adjust.）

**Comment by chenbolinstudent** [comment]
how about the difference  of the Short Connections and the side output 1 2 3 4 5 6?

**Comment by chenbolinstudent** [comment]
when compare with the orginal paper ,how about the difference of the Short Connections , the side output 1 2 3 4 5 6 ?

**Comment by AceCoooool** [comment]
Follow picture:
![image](https://user-images.githubusercontent.com/16620320/48414995-1d1f6c80-e787-11e8-8d88-48c75cc81bee.png)

Inference：
![image](https://user-images.githubusercontent.com/16620320/48415037-43dda300-e787-11e8-8b66-785817800802.png)
And the paper use Z_fuse=2,3,4; however, in here, I use all the outputs

**Comment by chenbolinstudent** [comment]
what is the meaning ?
extra = {'dss': [(64, 128, 3, [8, 16, 32, 64]), (128, 128, 3, [4, 8, 16, 32]), (256, 256, 5, [8, 16]),
                 (512, 256, 5, [4, 8]), (512, 512, 5, []), (512, 512, 7, [])]}
connect = {'dss': [[2, 3, 4, 5], [2, 3, 4, 5], [4, 5], [4, 5], [], []]}
k = [out[x] for x in [1, 2, 3, 6]]
thanks
